packages/aws-cdk-lib/aws-appsync/lib/source-api-association.ts

```diff
@@ -0,0 +1,220 @@
+   */
+  public static fromSourceApiAssociationAttributes(scope: Construct, id: string, attrs: SourceApiAssociationAttributes): ISourceApiAssociation {
+    class Import extends Resource {
+      public readonly associationId = Lazy.stringValue({
+        produce: () => Fn.select(3, Fn.split('/', attrs.associationArn)),
+      });
+
+      public readonly associationArn = attrs.associationArn;
+      public readonly sourceApi = attrs.sourceApi;
+      public readonly mergedApi = attrs.mergedApi;
+      constructor(s: Construct, i: string) {
+        super(s, i);
+      }
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * The association id.
+   */
+  readonly associationId: string;
+
+  /**
+   * The association arn.
+   */
+  readonly associationArn: string;
+
+  /**
+  * The underlying CFN source api association resource.
+  */
+  public readonly association: CfnSourceApiAssociation;
+
+  /**
+   * The merged api in the association.
+   */
+  public readonly mergedApi: IGraphqlApi;
+
+  /**
+   * The source api in the association.
+   */
+  public readonly sourceApi: IGraphqlApi;
+
+  /**
+  * The merge type for the source api association.
+  */
+  public readonly mergeType: MergeType;
+
+  /**
+  * The merged api execution role for attaching the access policy.
+  */
+  private readonly mergedApiExecutionRole?: IRole;
+
+  constructor(scope: Construct, id: string, props: SourceApiAssociationProps) {
+    super(scope, id);
+
+    this.mergeType = props.mergeType ?? MergeType.AUTO_MERGE;
+    this.mergedApiExecutionRole = props.mergedApiExecutionRole;
+    this.sourceApi = props.sourceApi;
+    this.mergedApi = props.mergedApi;
+
+    this.association = new CfnSourceApiAssociation(this, 'Resource', {
+      sourceApiIdentifier: this.sourceApi.arn,
+      mergedApiIdentifier: this.mergedApi.arn,
+      sourceApiAssociationConfig: {
+        mergeType: this.mergeType,
+
+      },
+      description: props.description,
+    });
+
+    this.associationId = this.association.attrAssociationId;
+    this.associationArn = this.association.attrAssociationArn;
+
+    // Add permissions to the merged api execution role if it was passed in.
+    if (this.mergedApiExecutionRole) {
+      addSourceGraphQLPermission(this.association, this.mergedApiExecutionRole);
+      if (this.mergeType === MergeType.AUTO_MERGE) {
+        addSourceApiAutoMergePermission(this.association, this.mergedApiExecutionRole);
+      }
+    }
+  }
+}
+
+/**
+* Adds an IAM permission to the Merged API execution role for GraphQL access on the source AppSync api.
+*
+* @param sourceApiAssociation The CfnSourceApiAssociation resource which to add a permission to access at runtime.
+* @param mergedApiExecutionRole The merged api execution role on which to add the permission.
+*/
+export function addSourceGraphQLPermission(sourceApiAssociation: CfnSourceApiAssociation, mergedApiExecutionRole: IRole) {
```

Comment:
Does this need to be a free floating function? Can it be a private method on `SourceApiAssociation`? Or even on `ISourceApiAssociation` since it operates on `CfnSourceApiAssociation` under the hood.

Reply:
I am using it as free floating because unfortunately this logic is needed by GraphqlApi construct as well and I want it used in one place. Should I make it a static within SourceApiAssociation?

Reply:
Discussed offline, keeping it free floating as its shared with GraphqlApi construct as well